src/commands/registerCommands.ts

```diff
@@ -57,66 +57,90 @@ import { pruneVolumes } from "./volumes/pruneVolumes";
-    registerCommand('vscode-docker.registries.azure.deleteRegistry', deleteAzureRegistry);
-    registerCommand('vscode-docker.registries.azure.deleteRepository', deleteAzureRepository);
-    registerCommand('vscode-docker.registries.azure.openInPortal', openInAzurePortal);
-    registerCommand('vscode-docker.registries.azure.runTask', runAzureTask);
-    registerWorkspaceCommand('vscode-docker.registries.azure.runFileAsTask', runFileAsAzureTask);
-    registerCommand('vscode-docker.registries.azure.selectSubscriptions', () => commands.executeCommand("azure-account.selectSubscriptions"));
-    registerCommand('vscode-docker.registries.azure.untagImage', untagAzureImage);
-    registerCommand('vscode-docker.registries.azure.viewProperties', viewAzureProperties);
-    registerCommand('vscode-docker.registries.azure.viewTaskLogs', viewAzureTaskLogs);
+    registerInternal(true, 'vscode-docker.registries.azure.buildImage', buildImageInAzure);
+    registerInternal(false, 'vscode-docker.registries.azure.createRegistry', createAzureRegistry);
+    registerInternal(false, 'vscode-docker.registries.azure.deleteRegistry', deleteAzureRegistry);
+    registerInternal(false, 'vscode-docker.registries.azure.deleteRepository', deleteAzureRepository);
+    registerInternal(false, 'vscode-docker.registries.azure.openInPortal', openInAzurePortal);
+    registerInternal(false, 'vscode-docker.registries.azure.runTask', runAzureTask);
+    registerInternal(true, 'vscode-docker.registries.azure.runFileAsTask', runFileAsAzureTask);
+    registerInternal(false, 'vscode-docker.registries.azure.selectSubscriptions', async () => await commands.executeCommand("azure-account.selectSubscriptions"));
+    registerInternal(false, 'vscode-docker.registries.azure.untagImage', untagAzureImage);
+    registerInternal(false, 'vscode-docker.registries.azure.viewProperties', viewAzureProperties);
+    registerInternal(false, 'vscode-docker.registries.azure.viewTaskLogs', viewAzureTaskLogs);
 
-    registerCommand('vscode-docker.volumes.configureExplorer', configureVolumesExplorer);
-    registerCommand('vscode-docker.volumes.inspect', inspectVolume);
-    registerCommand('vscode-docker.volumes.prune', pruneVolumes);
-    registerCommand('vscode-docker.volumes.remove', removeVolume);
+    registerInternal(false, 'vscode-docker.volumes.configureExplorer', configureVolumesExplorer);
+    registerInternal(false, 'vscode-docker.volumes.inspect', inspectVolume);
+    registerInternal(false, 'vscode-docker.volumes.prune', pruneVolumes);
+    registerInternal(false, 'vscode-docker.volumes.remove', removeVolume);
+}
+
+// tslint:disable-next-line: no-any
+function registerInternal<T>(workspace: boolean, commandId: string, callback: (context: IActionContext, ...args: any[]) => T | PromiseLike<T>, debounce?: number): void {
+    let registerFunction = workspace ? registerWorkspaceCommand : registerCommand;
+
+    registerFunction(
```

Comment:
@EricJizbaMSFT do you think this is an OK approach? Or might this mess up the registerCommand API?

Reply:
Can you refactor so that your new error is only shown when `ext.dockerode` is used? ENOENT is a pretty generic error type that can happen for a lot of reasons - also this new message would never make sense for registry commands or things like "Add Docker Files to Workspace"

Reply:
Is there an easier way to do that than just wrapping all the touches to ext.dockerode?

Reply:
Well I don't think you want to wrap _all_ touches because the tree will already display the better message:
![Screen Shot 2019-09-30 at 9 07 42 AM](https://user-images.githubusercontent.com/11282622/65896307-d2d4b900-e361-11e9-9e57-a0b98957e53f.png)

Looks like prune and anything using `osUtils.getDockerOSType` might be the only cases to handle

Reply:
Sounds good. I'll make it so.